plugins/catalog-backend-module-github/src/providers/GithubEntityProvider.ts

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TaskRunner } from '@backstage/backend-tasks';
+import { Config } from '@backstage/config';
+import {
+  GithubCredentialsProvider,
+  ScmIntegrations,
+  GitHubIntegrationConfig,
+  DefaultGithubCredentialsProvider,
+  SingleInstanceGithubCredentialsProvider,
+} from '@backstage/integration';
+import {
+  EntityProvider,
+  EntityProviderConnection,
+  LocationSpec,
+  locationSpecToLocationEntity,
+} from '@backstage/plugin-catalog-backend';
+
+import { graphql } from '@octokit/graphql';
+import * as uuid from 'uuid';
+import { Logger } from 'winston';
+import { camelCase } from 'lodash';
+import { getOrganizationEntities } from '../lib/github';
+
+/**
+ * Options for {@link GitHubEntityProvider}.
+ *
+ * @public
+ */
+export interface GitHubEntityProviderOptions {
+  /**
+   * A unique, stable identifier for this provider.
+   *
+   * @example "production"
+   */
+  id: string;
+
+  /**
+   * The target that this provider should consume.
+   *
+   * @example "https://github.com/backstage"
+   */
+  target: string;
+
+  /**
+   * An array of entity file names to be parsed and added to the catalog
```

Comment:
Let's explain more about how this works, I'm thinking pointing out that all of them will be added, and that in the case of a glob it's search all repos